core/vm/evm.go

```diff
@@ -149,13 +153,19 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig
 		chainConfig: chainConfig,
 		chainRules:  chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Random != nil, blockCtx.Time),
 	}
+	if txCtx.AccessEvents == nil && chainConfig.IsPrague(blockCtx.BlockNumber, blockCtx.Time) {
+		evm.AccessEvents = evm.StateDB.(*state.StateDB).NewAccessEvents()
+	}
 	evm.interpreter = NewEVMInterpreter(evm)
 	return evm
 }
 
 // Reset resets the EVM with a new transaction context.Reset
 // This is not threadsafe and should only be done very cautiously.
 func (evm *EVM) Reset(txCtx TxContext, statedb StateDB) {
+	if txCtx.AccessEvents == nil && evm.chainRules.IsPrague {
```

Comment:
It's a bit confusing for me. 

Should we always init a new `accessEvent` for each transaction? If so, we can do the initialization logic in `Statedb.Prepare` (and also we can decide whether the initialization is necessary by `rules.IsEIP4762`.

Besides, we can define a  `AccessEvent` function to expose the accessEvent from the statedb, so that EVM can manipulate it.

It feels very wonky to me to do the initialization here.